testing/sdk_utils.py

```diff
@@ -243,3 +242,61 @@ def merge_dictionaries(dict1: Dict, dict2: Optional[Dict]) -> Dict:
         else:
             ret[k] = dict2[k]
     return ret
+
+
+def get_service_roles(service_name) -> dict:
+
+    # Get the current service state to verify roles have applied.
+    mesos_state = sdk_cmd.cluster_request("GET", "/mesos/master/state").json()
+
+    service_state = None
+    current_service_roles = {}
+
+    # Find our service.
+    for service in mesos_state["frameworks"]:
+        if service["name"] == service_name:
+            service_state = service
+
+    # Create a map of tasks to roles.
+    if service_state:
+
+        if "roles" in service_state:
+            # MUTI_ROLE
+            current_service_roles["framework-roles"] = service_state["roles"]
+            current_service_roles["framework-role"] = None
+        else:
+            # not MULTI_ROLE
+            current_service_roles["framework-roles"] = None
+            current_service_roles["framework-role"] = service_state["role"]
+
+        current_service_roles["task-roles"] = {}
+        current_tasks = service_state["tasks"]
+        for task in current_tasks:
+            current_service_roles["task-roles"][task["name"]] = task["role"]
+
+    return current_service_roles
+
+
+def filter_role_from_config(unfiltered_config: Optional[Dict[str, Any]]) -> None:
+    if unfiltered_config is None:
+        return
+
+    # Remove any role related fields as role changes
+    # are allowed to happen across config targets
+    # to accomodate for quota migration.
+
+    # Inspired by:
+    # https://stackoverflow.com/a/50444005/10840685
```

Comment:
ah, TIL! 

i remembered the scala collection api's `flatten` method which can be used to accomplish this in a much simpler way.